### DISCUSSION TITLE
handleSubmit returns non-registered fields from defaultValues JSON

### DISCUSSION THREAD

**Discussion by PraveenPeri** [react-hook-form/react-hook-form#7047] [Q&A] [answered]
I have a multi-step form with defaultValues json passed to it. The JSON contains all the fields appearing in all the steps. However, we show only a subset of those fields per step and they will be posted to the API as the user continues to the next step. We are using <Controller /> component to render Material UI components on our form. We noticed that the data returned to the handleSubmit() hook contains all the fields from all the steps instead of just the ones  on the current step. Our expectation was that since only the fields on the current step are registered with the form, the submit handler will only receive those fields but that's not the case. Am I understanding this wrong? What should we do to receive only the current step's fields in the handler?

Here's a repro from the [sandbox](https://codesandbox.io/s/react-hook-form-v7-controller-forked-o29mm). You can see that the defaultValues JSON contains the "_unregisteredField" that's not being rendered. However, when you click 'Submit' button, the JSON data(coming from handleSubmit()) displayed also contains the new field.

**Comment by bluebill1049** [comment, marked as answer]
replied here as well: https://github.com/react-hook-form/react-hook-form/issues/7048
